### ISSUE TITLE
BundleResource with a quote (%27) in its name causes a build error

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#20058]
Test case:

https://github.com/rolfbjarne/xamarin-macios/commit/f96c36e5eb74d5fc4bbc81b992a5e5a19f7b4c80

Error:

> tools/msbuild/iOS/Xamarin.Shared.targets(340,88): error MSB4113: Specified condition "$([MSBuild]::ValueOrDefault('%(Identity)', '').StartsWith('$(AppBundleDir)/Contents/'))" evaluates to "$([MSBuild]::ValueOrDefault('bin/Debug/net8.0-maccatalyst/maccatalyst-arm64/MySimpleApp.app/Contents/Resources/Rolf's Quote.txt', '').StartsWith('$(AppBundleDir)/Contents/'))" instead of a boolean.

Binlog:

[build-2024-02-07-154147.binlog.zip](https://github.com/xamarin/xamarin-macios/files/14195302/build-2024-02-07-154147.binlog.zip)

**Comment by CartBlanche** [comment]
Thanks @rolfbjarne !

**Comment by CartBlanche** [comment]
@rolfbjarne Can I assume this is **very** low priority because VS4Mac is disappearing, or will it be addressed soon because VSCode also ships on macOS?

**Comment by rolfbjarne** [comment]
> @rolfbjarne Can I assume this is **very** low priority because VS4Mac is disappearing, or will it be addressed soon because VSCode also ships on macOS?

Neither?

It seems to me there's a simple workaround: don't use filenames with quotes, or is there a specific reason that doesn't work for you?

**Comment by CartBlanche** [comment]
@rolfbjarne well that same audio file works in Xamarin Android and Xamarin Windows, with zero need for changes and the customer's content references that file specifically. There are 30-40 audio files with an apostrophe in the name. Maybe more.

As the code is shared across Xamarin.Forms projects it would be annoying to have an exception just for MacOS applications. This was previously working in older versions of Xamarin.Mac, so this is a regression after updating the project, after about 2 years. I obviously don't know in which version it broke.

**Comment by CartBlanche** [comment]
@rolfbjarne I'm obviously no MSBuild guru, but would something like this at line 340 in `Xamarin.Shared.targets`:
`<_FilesInAppBundleRootDirectoryToRemove Include="@(_FilesInAppBundleRootDirectory)"
                                      Condition="$([MSBuild]::ValueOrDefault('%(Identity)', '').Replace("'", "").StartsWith('$(AppBundleDir)/Contents/'))" />`
fix this issue, or would it cause other problems?

**Comment by rolfbjarne** [comment]
Adding this to your csproj might work:

```xml
<Target Name="_CleanAppBundleRootDirectory" />
```

it just removes the target that causes the problem.

Can you try and see if it works for you?

**Comment by CartBlanche** [comment]
@rolfbjarne should that actually be something like:
```
<Target Name="_CleanAppBundleRootDirectory">
  <Remove Dir="$(AppBundleRootDirectory)" />
</Target>
```

I just want to make sure I'm putting the right thing in the csproj.

**Comment by rolfbjarne** [comment]
@CartBlanche no, the target can be empty, it doesn't do anything all that important (it's just preventing a possible confusing codesigning error if there's extraneous files in the app bundle when the app bundle is signed - worst case you'll end up confused because of strange codesigning errors).

**Comment by CartBlanche** [comment]
@rolfbjarne it doesn't get rid of the error. But I just noticed the actual executable, in Debug mode at Least, is created and runs, without that change. Tomorrow I'll see if I can build a Release version and sign it.

**Comment by CartBlanche** [comment]
@rolfbjarne Btw, when I say runs, if doesn't run from the IDE, I have to manually run the exe, due to the build error stopping progress. Today I could build Release builds, but any projects that have the apostrophe error will not build a *.pkg file, nor will it allow archive for publishing, so I cannot sign them, due to the regression build error.
